DataFormats/JetReco/BuildFile.xml

```diff
@@ -6,6 +6,7 @@
 <use name="DataFormats/ParticleFlowReco"/>
 <use name="DataFormats/TrackReco"/>
 <use name="FWCore/Utilities"/>
+<use name="DataFormats/Candidate"/>
```

Comment:
Looking at the UBSAN IBs log you are pointing at in the PR description I was able to find the build errors that require the updates you are proposing for the other three BuildFile's, but not for this one.

For my own understending, could you please point me to the error in the log that suggests adding "DataFormats/Candidate" here?

Reply:
I think the issue was that `DQMOffline/Alignment` had build error [a] but I noticed that it is getting `LeafCandidate` via `DataFormats/JetReco` with out dependency in BuildFile[b]. So I added the required dependency for `DQMOffline/Alignment` ( https://github.com/cms-sw/cmssw/pull/30023/files#diff-f6befb70f0da8d2364ca45ff83e014a4R7 ) but error was still not fixed. Then I noticed that  `DataFormats/JetReco` uses `DataFormats/Candidate` without explicit dep in BuildFile. That is why I proposed this change.

[a]
https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/slc7_amd64_gcc820/CMSSW_11_2_UBSAN_X_2020-05-22-2300/DQMOffline/Alignment
```
undefined reference to `typeinfo for reco::LeafCandidate'
```
[b]
```
DQMOffline/Alignment/interface/TkAlCaRecoMonitor.h:#include <DataFormats/JetReco/interface/CaloJet.h>
```

Reply:
by bad here, looks like DataFormats/JetReco already has the right deps. I will revert this change. please hold